renderer.js

```diff
@@ -1713,9 +1771,41 @@ class SDFGRenderer {
             // viewport and tell it to re-sort the shown transformations.
             try {
                 if (vscode) {
+                    function clean_selected(selected_elements) {
+                        let elems = [];
+                        selected_elements.forEach((el) => {
+                            console.log(el);
+                            if (el.data.node)
+                                elems.push({
+                                    type: 'node',
+                                    sdfg_id: el.sdfg.sdfg_list_id,
+                                    state_id: el.parent_id,
+                                    id: el.id,
+                                });
+                            else if (el.data.state)
+                                elems.push({
+                                    type: 'state',
+                                    sdfg_id: el.sdfg.sdfg_list_id,
+                                    state_id: -1,
+                                    id: el.id,
+                                });
+                            else
+                                elems.push({
+                                    type: 'edge',
+                                    sdfg_id: el.sdfg.sdfg_list_id,
+                                    state_id: -1,
```

Comment:
No distinction between memlets and interstate edges. (some of them have state_id != -1)